Controller/CRUDController.php

```diff
@@ -63,6 +63,9 @@ public function setContainer(ContainerInterface $container = null)
      */
     public function render($view, array $parameters = array(), Response $response = null)
     {
+        if (!$this->isXmlHttpRequest()) {
+            $parameters['breadcrumbs_builder'] = $this->get('sonata.admin.breadcrumbs_builder');
+        }
```

Comment:
> I'm talking about twig, not C/C++. :stuck_out_tongue: 

Lol most of it still applies : 

-  Non-locality : you get the global variable in every template, even if you don't need it
- No Access Control or Constraint Checking : every template can fuck with the global variable
- Implicit coupling : you don't "see" that your template is bound to the variable
- Namespace pollution : that's one less name you can use